lib/ecom/api.ts

```diff
@@ -282,6 +285,46 @@ function createEcomApi(wixClient: WixApiClient): EcomAPI {
                 return failureResponse(EcomApiErrorCodes.GetProductsFailure, getErrorMessage(e));
             }
         },
+        isUserLoggedIn() {
+            return wixClient.auth.loggedIn();
+        },
+        async getUser() {
+            const response = await wixClient.members.getCurrentMember();
+            return response.member;
+        },
+        async login(callbackUrl: string, returnUrl: string) {
+            const oAuthData = wixClient.auth.generateOAuthData(callbackUrl, returnUrl);
+
+            const { authUrl } = await wixClient.auth.getAuthUrl(oAuthData, {
+                responseMode: 'query',
+            });
+
+            return { oAuthData, authUrl };
+        },
+        async logout(returnUrl: string) {
+            const result = await wixClient.auth.logout(returnUrl);
+            return result;
+        },
+        async handleLoginCallback(url, oAuthData) {
```

Comment:
this method feels out of place here... maybe it's better to have to separate methods: `getMemberTokens` and `setMemberTokens`?

Reply:
we also using `wixClient.auth.parseFromUrl(url, 'query');` for this.
so, it would be like:

```
const api = initializeEcomApi();

const oAuthData = api.parseOauthDataFromUrl(request.url);

const originalOauthData = session.get("oAuthData");

const memberTokens = api.getMemberTokens(oAuthData, originalOAuthData);

api.setMemberTokens(memberTokens)

```

I don't think this makes a lot of sense. in that case, instead of incapsulate logic inside single place we are spreading it across application

Reply:
but what does `handleLoginCallback` (don't even look at the implementation) have to do with the API? Maybe it uses API but it should not be part of the API.
Why not having it in place where it's used: 
```
const handleLoginCallback = () => {
   ... // the code from your comment here
}
```

> instead of incapsulate logic inside single place we are spreading it across application

in my opinion it's not spreading it across the application, it's separation of concerns.. sorry for my academic background 😄

Reply:
refactored as discussed